substrate/frame/staking/src/migrations.rs

```diff
@@ -59,11 +60,42 @@ impl Default for ObsoleteReleases {
 #[storage_alias]
 type StorageVersion<T: Config> = StorageValue<Pallet<T>, ObsoleteReleases, ValueQuery>;
 
+/// Migrating `OffendingValidators` from `Vec<(u32, bool)>` to `Vec<u32>`
+pub mod v15 {
+	use super::*;
+
+	pub struct VersionUncheckedMigrateV14ToV15<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> OnRuntimeUpgrade for VersionUncheckedMigrateV14ToV15<T> {
+		fn on_runtime_upgrade() -> Weight {
+			let migrated = v14::OffendingValidators::<T>::get()
+				.into_iter()
+				.map(|p| p.0)
+				.collect::<Vec<_>>();
+			DisabledValidators::<T>::set(migrated);
+
+			log!(info, "v15 applied successfully.");
+			T::DbWeight::get().reads_writes(1, 1)
```

Comment:
as per discussed in https://github.com/paritytech/polkadot-sdk/pull/3640, this code is kinda wrong if ever used in parachains. 

Won't ask you to change or anything, but I think it is important to learn why :)